src/autopas/containers/linkedCells/traversals/LCC08CellHandlerUtility.h

```diff
@@ -0,0 +1,150 @@
+  /** The cell diagonally to the origin **/
+  BACK_RIGHT,
+};
+
+/**
+ * Array containing all four enum values of {@link autopas::internal::C08CellDirection}.
+ */
+constexpr inline std::array<C08CellDirection, 4> ALL_DIRECTIONS{
+    {C08CellDirection::FRONT_LEFT, C08CellDirection::BACK_LEFT, C08CellDirection::FRONT_RIGHT,
+     C08CellDirection::BACK_RIGHT}};
+
+/**
+ * Calculates the offset for cell2 in the pair, given a direction from the base cell, a z index and the overlap.
+ * @param overlap1 the overlap of the interacting cells (plus one)
+ * @param direction one of the four directions
+ * @param z the z index of the spatial dimension
+ * @return offset of cell2
+ */
+constexpr int calculateCell2Index(int overlap1, const C08CellDirection &direction, int z);
+
+/**
+ * Calculates the multipliers (zero or one) for a given direction.
+ * This is used as a helper to calculate the `distVec` in {@link autopas::internal::computePairwiseCellOffsetsC08}.
+ * @param direction one of the four directions
+ * @return pair of multipliers: first one for x dimension, second one for y dimension
+ */
+constexpr std::pair<int, int> toDistVectorMultiplier(const C08CellDirection &direction);
+
+/**
+ * Returns true if the cell-interaction in the given direction, with the given base cell coordinates and overlap
+ * needs to be included into the pairwise cell offsets
+ * @param direction the direction from base cell to inetracting cell
+ * @param overlap the overlap (calculated from interactionLength divided by cellLength)
+ * @param x the x offset of the base cell
+ * @param y the y offset of the base cell
+ * @param z the z offset of the base cell
+ * @return true if the pair needs to be included
+ */
+constexpr bool includeC08Case(const C08CellDirection &direction, const std::array<int, 3> &overlap, int x, int y,
+                              int z);
+
+/**
+ * Computes the sorting direction between two cells from center of cell1 to center of cell2.
+ * @param cellsPerDimension array containing the number of cells per dimension
+ * @param distVec1 the vector to cell1 from the origin
+ * @param offset2 the offset of cell2
+ * @return a vector containing the sorting direction, i.e. the vector from center of cell1 to center of cell2
+ */
+std::array<double, 3> computeSortingDirection(const std::array<int, 3> &cellsPerDimension,
+                                              const std::array<double, 3> &distVec1, int offset2);
+
+/**
+ * Computes the cell offsets for the C08 base step based on the given number of cells per dimension and overlap.
+ * @param cellsPerDimension the number of cells per dimension
+ * @param overlap the extent of overlap between neighboring cells
+ * @return a vector of unsigned long values representing the computed cell offsets in 1D coordinates
+ */
+std::vector<int> computeCellOffsetsC08(const std::array<unsigned long, 3> &cellsPerDimension,
+                                       const std::array<int, 3> &overlap);
```

Comment:
The function name and doc feels slightly misleading. It only gives us the offsets (or relative indices) of all cells in the region of the c08 base step. This does not yet describe the base step itself.

Reply:
Agreed